services/ens/api.go

```diff
@@ -54,10 +58,18 @@ type API struct {
 	accountsManager *account.GethManager
 	rpcFiltersSrvc  *rpcfilters.Service
 	config          *params.NodeConfig
+
+	addrPerChain      map[uint64]common.Address
+	addrPerChainMutex sync.Mutex
+	quit              chan struct{}
```

Comment:
would ` quit chan int` works? seems simpler

Reply:
It would work. I went for `chan struct{}` because it is a common pattern to use for signal channel, due to it not allocating memory for a value.